app/assets/v2/js/cart-data.js

```diff
@@ -246,6 +247,74 @@ class CartData {
+      const latestGrantData = latestGrantsData[grantIndex];
+
+      // Update with the grant data from server
+      cartData[index].binance_payout_address = latestGrantData.binance_payout_address;
+      cartData[index].celo_payout_address = latestGrantData.celo_payout_address;
+      cartData[index].clr_round_num = latestGrantData.clr_round_num;
+      cartData[index].grant_admin_address = latestGrantData.admin_address;
+      cartData[index].grant_clr_prediction_curve = latestGrantData.clr_prediction_curve;
+      cartData[index].grant_contract_address = latestGrantData.contract_address;
+      cartData[index].grant_contract_version = latestGrantData.contract_version;
+      cartData[index].grant_donation_num_rounds = 1; // always 1, since recurring contributions are not supported
+      cartData[index].grant_id = String(latestGrantData.id); // IDs are saved as strings in localStorage, so we cast to string here
+      cartData[index].grant_image_css = latestGrantData.image_css;
+      cartData[index].grant_logo = latestGrantData.logo_url;
+      cartData[index].grant_slug = latestGrantData.slug;
+      cartData[index].grant_title = latestGrantData.title;
+      cartData[index].grant_token_address = latestGrantData.token_address;
+      cartData[index].grant_token_symbol = latestGrantData.token_symbol;
+      cartData[index].grant_url = latestGrantData.url;
+      cartData[index].harmony_payout_address = latestGrantData.harmony_payout_address;
+      cartData[index].is_clr_eligible = latestGrantData.is_clr_eligible;
+      cartData[index].is_on_team = latestGrantData.is_on_team;
+      cartData[index].kusama_payout_address = latestGrantData.kusama_payout_address;
+      cartData[index].polkadot_payout_address = latestGrantData.polkadot_payout_address;
+      cartData[index].rsk_payout_address = latestGrantData.rsk_payout_address;
+      cartData[index].tenants = latestGrantData.tenants;
+      cartData[index].zcash_payout_address = latestGrantData.zcash_payout_address;
+      cartData[index].zil_payout_address = latestGrantData.zil_payout_address;
+
+      // Other localStorage properties not updated above:
+      //   grant_donation_amount:     Existing localStorage value is preserved
+      //   grant_donation_clr_match:  Updated in cart.js `grantData` watcher
+      //   grant_donation_currency:   Existing localStorage value is preserved
+      //   payment_status:            Updated on-demand in cart.js `updatePaymentStatus()` method
+      //   token_local_id:            Existing localStorage value is preserved
+      //   txnid:                     Updated on-demand in cart.js `updatePaymentStatus()` method
+      //   uuid:                      TODO is this needed or can it be removed? Currently existing value is preserved
+    });
+
+    // Save the updated cart data to local storage and return it
+    this.setCart(cartData);
+    return cartData;
```

Comment:
If we do `CartData.addToCart(grant)`, won't that overwrite a user's `grant_donation_amount` and `grant_donation_currency` that are already saved in localstorage? I believe it would, which is why I did it this way as indicated in the comments here

Reply:
check cart-data.js addToCart it only overwrite if not present
![image](https://user-images.githubusercontent.com/1358093/110863303-40974b00-829f-11eb-87d4-3b3cf13fdb2c.png)